Configuration mismatch between docs and code

The XML configuration in the docs (even in Fabric8 ones), does not seem to match what the code does. This was a source of confusion when I started using these plugins initially (until I dug through the source code).

For example, the [XML Configuration](https://htmlpreview.github.io/?https://github.com/eclipse/jkube/blob/master/kubernetes-maven-plugin/doc/index.html#xml-configuration) outlines what `<resources>` item looks like but this does not match the [`ResourceConfig`](https://github.com/eclipse/jkube/blob/master/jkube-kit/config/resource/src/main/java/org/eclipse/jkube/kit/config/resource/ResourceConfig.java) class.

If I provide say the `<deployment>` child as outlined in that section, I get the following error:

```
[ERROR] Failed to execute goal org.eclipse.jkube:k8s-maven-plugin:0.1.1:resource (default-cli) on project pdfg: Unable to parse configuration of mojo org.eclipse.jkube:k8s-maven-plugin:0.1.1:resource for parameter deployment: Cannot find 'deployment' in class org.eclipse.jkube.kit.config.resource.ResourceConfig -> [Help 1]
```

But when I do follow the class structure, everything goes as expected.

Am I missing something here?

--- a/jkube-kit/config/resource/src/main/java/org/eclipse/jkube/kit/config/resource/ServiceAccountConfig.java
+++ b/jkube-kit/config/resource/src/main/java/org/eclipse/jkube/kit/config/resource/ServiceAccountConfig.java
@@ -19,16 +19,47 @@ public class ServiceAccountConfig {
 
     private String deploymentRef;
 
-    public ServiceAccountConfig(String name) {
-        this.name = name;
-    }
-
     public String getName() {
         return name;
     }
 
     public String getDeploymentRef() {
         return deploymentRef;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setDeploymentRef(String deploymentRef) {
+        this.deploymentRef = deploymentRef;
+    }
+
+    public static class Builder {
+        private ServiceAccountConfig serviceAccountConfig = new ServiceAccountConfig();
+
+        public Builder() { }
+
+        public Builder(ServiceAccountConfig serviceAccountConfig) {
+            if (serviceAccountConfig != null) {
+                this.serviceAccountConfig.name = serviceAccountConfig.getName();
+                this.serviceAccountConfig.deploymentRef = serviceAccountConfig.getDeploymentRef();
+            }
+        }
+
+        public Builder withName(String name) {
+            this.serviceAccountConfig.name = name;
+            return this;
+        }
+
+        public Builder withDeploymentRef(String deploymentRef) {
+            this.serviceAccountConfig.deploymentRef = deploymentRef;
+            return this;
+        }
+
+        public ServiceAccountConfig build() {
+            return serviceAccountConfig;
+        }
     }
 }
 

--- a/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ServiceAccountEnricherTest.java
+++ b/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ServiceAccountEnricherTest.java
@@ -44,7 +44,7 @@ public class ServiceAccountEnricherTest {
             context.getConfiguration();
             result = new Configuration.Builder()
                     .resource(new ResourceConfig.Builder()
-                            .withServiceAccounts(Collections.singletonList(new ServiceAccountConfig("ribbon"))).build())
+                            .withServiceAccounts(Collections.singletonList(new ServiceAccountConfig.Builder().withName("ribbon").build())).build())
                     .build();
         }};
         final KubernetesListBuilder builder = new KubernetesListBuilder();

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_access.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_access.adoc
@@ -16,4 +16,6 @@ be specified by the certPath or machine configuration, or by the
 
 == OpenShift and Kubernetes Access
 
-If no `DOCKER_HOST` is set and no unix socket could be accessed under `/var/run/docker.sock` then f-m-p checks whethe https://github.com/jkubeio/gojkube.gojkube. is in the path and uses `gojkube.docker-env` to get the connection parameter to the Docker host exposed by
+Plugin reads your `kubeconfig` file to read your Kubernetes/OpenShift configuration. By default, `kubeconfig` file is assumed to be either
+in `~/.kube/config` or using the environment variable `KUBECONFIG`.
+

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_compatibility.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_compatibility.adoc
@@ -9,12 +9,12 @@
 |===
 |     KMP             | Kubernetes 1.17.0 | Kubernetes 1.14.2 | Kubernetes 1.12.2 | Kubernetes 1.11.0 | Kubernetes 1.10.0 | Kubernetes 1.9.0 | Kubernetes 1.8.0 | Kubernetes 1.7.0 | Kubernetes 1.6.0 | Kubernetes 1.5.1 | Kubernetes 1.4.0
 
-| KMP 1.0.0-alpha-1   |        ✓          |        ✓          |        ✓          |        ✓          |        ✓          |        ✓         |        ✓         |        ✓        |        ✓         |        ✓         |        ✓        |
+| KMP 1.0.0-alpha-1   |        ✓          |        ✓          |        ✓          |        ✓          |        ✓          |        ✓         |        ✓         |        ✓        |        ✓         |        ✓         |        ✓        
 
-| KMP 0.2.0           |        ✓          |        ✓          |        ✓          |        ✓          |        ✓          |        ✓         |        ✓         |        ✓        |        ✓         |        ✓         |        ✓        |
+| KMP 0.2.0   |        ✓          |        ✓          |        ✓          |        ✓          |        ✓          |        ✓         |        ✓         |        ✓        |        ✓         |        ✓         |        ✓
 
-| KMP 0.1.1           |        x          |        ✓          |        ✓          |        ✓          |        ✓          |        ✓         |        ✓         |        ✓        |        ✓         |        ✓         |        ✓        |
+| KMP 0.1.1   |        x          |        ✓          |        ✓          |        ✓          |        ✓          |        ✓         |        ✓         |        ✓        |        ✓         |        ✓         |        ✓
 
-| KMP 0.1.0           |        x          |        ✓          |        ✓          |        ✓          |        ✓          |        ✓         |        ✓         |        ✓        |        ✓         |        ✓         |        ✓        |
+| KMP 0.1.0   |        x          |        ✓          |        ✓          |        ✓          |        ✓          |        ✓         |        ✓         |        ✓        |        ✓         |        ✓         |        ✓
 
 |===

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_enricher.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_enricher.adoc
@@ -308,10 +308,10 @@ These values can be configured by the enricher in the `kubernetes-maven-plugin` 
 <plugin>
   <groupId>org.eclipse.jkube</groupId>
   <artifactId>kubernetes-maven-plugin</artifactId>
-  <version>3.3.0</version>
+  <version>{version}</version>
   <executions>
     <execution>
-      <id>fmp</id>
+      <id>kmp</id>
         <goals>
           <goal>resource</goal>
           <goal>helm</goal>
@@ -364,10 +364,10 @@ These values can be configured by the enricher in the `kubernetes-maven-plugin` 
       <plugin>
         <groupId>org.eclipse.jkube</groupId>
         <artifactId>kubernetes-maven-plugin</artifactId>
-        <version>3.3.0</version>
+        <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -416,10 +416,10 @@ These values can be configured by the enricher in the `kubernetes-maven-plugin` 
       <plugin>
         <groupId>org.eclipse.jkube</groupId>
         <artifactId>kubernetes-maven-plugin</artifactId>
-        <version>3.3.0</version>
+        <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -468,10 +468,10 @@ These values can be configured by the enricher in the `kubernetes-maven-plugin` 
       <plugin>
         <groupId>org.eclipse.jkube</groupId>
         <artifactId>kubernetes-maven-plugin</artifactId>
-        <version>3.3.0</version>
+        <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -522,7 +522,7 @@ The minimal configuration to add health checks is the following:
         <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -569,7 +569,7 @@ You can provide two different configuration for the readiness and liveness check
         <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -614,7 +614,7 @@ Shared (generic) configuration can be set outside of the specific configuration.
         <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_introduction.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/_introduction.adoc
@@ -144,7 +144,7 @@ mvn k8s:resource k8s:deploy
 
 By default a _Service_ and a _Deployment_ object pointing to the created Docker image is created. When running in OpenShift mode, a _Service_ and _DeploymentConfig_ which refers the _ImageStream_ created with `k8s:build` will be installed.
 
-Of course you can bind all those jkube.goals to execution phases as well, so that they are called along with standard lifecycle goals like `install`. For example, to bind the building of the Kubernetes resource files and the Docker images, add the following goals to the execution of the f-m-p:
+Of course you can bind all those jkube.goals to execution phases as well, so that they are called along with standard lifecycle goals like `install`. For example, to bind the building of the Kubernetes resource files and the Docker images, add the following goals to the execution of the k-m-p:
 
 .Example for lifecycle bindings
 [source, xml, indent=0]
@@ -204,7 +204,7 @@ The plugin configuration can be roughly divided into the following sections:
 * `<generator>` configures <<generators,generators>> which are responsible for creating images. Generators are used as an alternative to a dedicated `<images>` section.
 * `<enricher>` configures various aspects of <<enrichers,enrichers>> for creating or enhancing resource descriptors.
 
-A working example can be found in the  https://github.com/jkubeio/kubernetes-maven-plugin/tree/master/samples/xml-config[samples/xml-config] directory. An extract of the plugin configuration is shown below:
+A working example can be found in the  https://github.com/eclipse/jkube/tree/master/kubernetes-maven-plugin/tree/master/quickstarts/maven/xml-config[quickstarts/maven/xml-config] directory. An extract of the plugin configuration is shown below:
 
 .Example for an XML configuration
 [source,xml,indent=0,subs="verbatim,quotes,attributes"]
@@ -217,10 +217,11 @@ A working example can be found in the  https://github.com/jkubeio/kubernetes-mav
       <!-- "alias" is used to correlate to the containers in the pod spec -->
       <alias>camel-app</alias>
       <build>
-        <from>jkube.java</from>
+        <from>fabric8/java-centos-openjdk8-jre</from>
         <assembly>
-          <basedir>/deployments</basedir>
-          <descriptorRef>artifact-with-dependencies</descriptorRef>
+           <inline>
+              <baseDirectory>/deployments</baseDirectory>
+           </inline>
         </assembly>
         <env>
           <JAVA_LIB_DIR>/deployments</JAVA_LIB_DIR>
@@ -237,31 +238,10 @@ A working example can be found in the  https://github.com/jkubeio/kubernetes-mav
       </all>
     </labels>
 
-    <deployment> <!--4-->
-      <name>${project.artifactId}</name>
-      <replicas>1</replicas>
+    <replicas>2</replicas> <!--4-->
+    <controllerName>${project.artifactId}</controllerName> <!--5-->
 
-      <containers> <!--5-->
-        <container>
-          <alias>camel-app</alias> <!--6-->
-          <ports>
-            <port>8778</port>
-          </ports>
-          <mounts>
-            <scratch>/var/scratch</scratch>
-          </mounts>
-        </container>
-      </containers>
-
-      <volumes> <!--7-->
-        <volume>
-          <name>scratch</name>
-          <type>emptyDir</type>
-        </volume>
-      </volumes>
-    </deployment>
-
-    <services> <!--8-->
+    <services> <!--6-->
       <service>
         <name>camel-service</name>
         <headless>true</headless>
@@ -279,11 +259,9 @@ A working example can be found in the  https://github.com/jkubeio/kubernetes-mav
 <1> Standard docker-maven-plugin configuration for building one single Docker image
 <2> Kubernetes / OpenShift resources to create
 <3> Labels which should be applied globally to all resource objects
-<4> Definition of a http://kubernetes.io/docs/user-guide/deployments/[Deployment] to create
-<5> Containers to include in the deployment
-<6> An _alias_ is used to correlate a container's image with the image definition in the `<images>` section where each image carries an alias. Can be omitted if only a single image is used
-<7> http://kubernetes.io/docs/user-guide/volumes/[Volume] definitions used in a Deployment's _ReplicaSet_
-<8> One or more http://kubernetes.io/docs/user-guide/services/[Service] definitions.
+<4> Number of replicas desired
+<5> Name of controller created by plugin
+<6> One or more http://kubernetes.io/docs/user-guide/services/[Service] definitions.
 
 The XML resource configuration is based on plain Kubernetes resource objects. When targeting OpenShift, Kubernetes resource descriptors will be automatically converted to their OpenShift counterparts, e.g. a Kubernetes http://kubernetes.io/docs/user-guide/deployments/[Deployment] will be converted to an OpenShift https://docs.openshift.org/latest/architecture/core_concepts/deployments.html#deployments-and-deployment-configurations[DeploymentConfig].
 
@@ -293,7 +271,7 @@ The third configuration option is to use an external configuration in form of YA
 
 Note: In order to support simultaneously both OpenShift and Kubernetes, there is currently no way to specify OpenShift-only features this way, though this might change in future releases.
 
-Let's have a look at an example from https://github.com/rhuss/kubernetes-maven-plugin/tree/master/samples/external-resources[samples/external-resources]. This is a plain Spring Boot application, whose images are auto generated like in the <<zero-config, Zero-Config>> case. The resource fragments are in `src/main/jkube`.
+Let's have a look at an example from https://github.com/eclipse/jkube/tree/master/quickstarts/maven/external-resources[quickstarts/maven/external-resources]. This is a plain Spring Boot application, whose images are auto generated like in the <<zero-config, Zero-Config>> case. The resource fragments are in `src/main/jkube`.
 
 .Example fragment "deployment.yml"
 [source,yaml,indent=0,subs="verbatim,quotes,attributes"]

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/generator/_java_exec.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/generator/_java_exec.adoc
@@ -22,9 +22,9 @@ It will use the following base image by default, but as explained <<generator-op
 | | Docker Build | S2I Build | ImageStream
 
 | *Community*
-| `jkube.java-jboss-openjdk8-jdk`
-| `jkube.s2i-java`
-| `jkube.java`
+| `fabric8/java-jboss-openjdk8-jdk`
+| `fabric8/s2i-java`
+| `fabric8-java`
 
 | *Red Hat*
 | `jboss-fuse-6/fis-java-openshift`

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-apply.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-apply.adoc
@@ -8,3 +8,82 @@ This goal applies the resources created with <<k8s:resource>> to a connected Kub
 ----
 mvn k8s:apply
 ----
+
+[[Supported-Properties-Apply]]
+=== Supported Properties For Apply goal
+
+.Other options available with apply goal
+[cols="1.6.3"]
+|===
+| Element | Description | Property(System property or maven property)
+
+| *failOnError*
+| Should we fail the build if an apply fails, by default `true`
+| `jkube.deploy.failOnError`
+
+| *recreate*
+| Should we update resources by deleting them first and then creating them again, by default `false`
+| `jkube.recreate`
+
+| *kubernetesManifest*
+| The generated kubernetes YAML file, defaults to `${basedir}/target/classes/META-INF/jkube/kubernetes.yml`
+| `jkube.kubernetesManifest`
+
+| *create*
+| Should we create new resources, defaults to `true`
+| `jkube.deploy.create`
+
+| *rolling*
+| Should we use rolling updates to apply changes, defaults to `false`
+| `jkube.rolling`
+
+| *failOnNoKubernetesJson*
+| Should we fail if there is no Kubernetes JSON, defaults to `false`
+| `jkube.deploy.failOnNoKubernetesJson`
+
+| *servicesOnly*
+| In services only mode we only process services so that those can be recursively created/updated first before creating/updating any pods and replication controllers, defaults to `false`
+| `jkube.deploy.servicesOnly`
+
+| *ignoreServices*
+| Do we want to ignore services. This is particularly useful when in recreate mode to let you easily recreate all the ReplicationControllers and Pods but leave any service definitions alone to avoid changing the portalIP addresses and breaking existing pods using the service. Defaults to `false`
+| `jkube.deploy.ignoreServices`
+
+| *processTemplatesLocally*
+| Process templates locally in Java so that we can apply OpenShift templates on any Kubernetes environment, defaults to `false`
+| `jkube.deploy.processTemplatesLocally`
+
+| *deletePods*
+| Should we delete all the pods if we update a Replication Controller, defaults to `true`
+| `jkube.deploy.deletePods`
+
+| *ignoreRunningOAuthClients*
+| Do we want to ignore OAuthClients which are already running?. OAuthClients are shared across namespaces so we should not try to update or create/delete global oauth clients, defaults to `true`
+| `jkube.deploy.ignoreRunningOAuthClients`
+
+| *createExternalUrls*
+| Should we create external Ingress for any LoadBalancer Services which don't already have them. Defaults to `false`
+| `jkube.deploy.createExternalUrls`
+
+| *jsonLogDir*
+| The folder we should store any temporary json files or results. Defaults to `${basedir}/target/jkube/applyJson`
+| `jkube.deploy.jsonLogDir`
+
+| *waitSeconds*
+| How many seconds to wait for a URL to be generated for a service , defaults to `5`
+| `jkube.serviceUrl.waitSeconds`
+
+| *resourceDir*
+| Folder where to find project specific files, defaults to `${basedir}/src/main/jkube`
+| `jkube.resourceDir`
+
+| *environment*
+|  Environment name where resources are placed. For example, if you set this property to dev and resourceDir is the default one, jkube will look at `src/main/jkube/dev`. Defaults to `null`
+| `jkube.environment`
+
+| *skipApply*
+| Skip applying the resources, defaults to `false`
+| `jkube.skip.apply`
+|===
+
+[[Supported-Properties]]

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc
@@ -62,7 +62,7 @@ Labels and annotations can be easily added to any resource object. This is best 
 <3> `<service>` labels are used to label services
 <4> `<replicaSet>` labels are for replica set and replication controller
 <5> `<pod>` holds labels for pod specifications in replication controller, replica sets and deployments
-<6> `<deployment>` is for labels on deployments (kubernetes) and deployment configs (openshift)
+<6> `<deployment>` is for labels on deployments (kubernetes)
 <7> The subelements are also available for specifying annotations.
 <8> `<remotes>` you can set location of fragments as `URL`.
 
@@ -76,10 +76,10 @@ The following subelements are possible for `<labels>` and `<annotations>` :
 | Element | Description
 
 | *all*
-| All entries specified in the `<all>` sections are applied to all resource objects created. This also implies build object like image stream and build configs which are created implicitely for an <<build-openshift, OpenShift build>>.
+| All entries specified in the `<all>` sections are applied to all resource objects created.
 
 | *deployment*
-| Labels and annotations applied to `Deployment` (for Kubernetes) and `DeploymentConfig` (for OpenShift) objects
+| Labels and annotations applied to `Deployment` (for Kubernetes)
 
 
 | *pod*
@@ -183,23 +183,6 @@ Resource goal also validates the generated resource descriptors using API specif
 | *jkube.failOnValidationError*
 | If value is set to `true` then any validation error will block the plugin execution. A warning will be printed otherwise.
 | `false`
-
-|===
-
-[[resource-route-generation]]
-=== Route Generation
-
-When the `k8s:resource` goal is run, an OpenShift https://docs.openshift.org/latest/architecture/networking/routes.html[Route] descriptor (`route.yml`) will also be generated along the service if an OpenShift cluster is targeted.
-If you do not want to generate a Route descriptor, you can set the `jkube.openshift.generateRoute` property to `false`.
-
-.Route Generation Configuration
-[cols="1.6.1"]
-|===
-| Configuration | Description | Default
-
-| *jkube.openshift.generateRoute*
-| If value is set to `false` then no Route descriptor will be generated. By default it is set to `true`, which will create a `route.yml` descriptor and also add Route resource to `openshift.yml`.
-| `true`
 |===
 
 If you do not want to generate a Route descriptor, you can also specify so in the plugin configuration in your POM as seen below.
@@ -235,46 +218,57 @@ spec:
 
 In case both the label and the property have been set with conflicting values, precedence will be given to the property value, so if you set the label to `true` but set the property to `false` then no Route descriptor will be generated because precedence will be given to the property value.
 
-[[Other-flags]]
-=== Other flags
+[[Supported-Properties-Resource]]
+=== Supported Properties for Resource goal
 
-.Other options available with resource goal
+.Options available with resource goal
 [cols="1.6.3"]
 |===
-| Configuration | Description | Default
+| Element | Description | Property(System property or maven property)
 
-| *jkube.profile*
-| Profile to use. A profile contains the enrichers and generators to use as well as their configuration. Profiles are looked up in the classpath and can be provided as yaml files.
-| `default`
+| *profile*
+| Profile to use. A profile contains the enrichers and generators to use as well as their configuration. Profiles are looked up in the classpath and can be provided as yaml files. Defaults to `default`
+| `jkube.profile`
 
-| *jkube.sidecar*
-| Whether to enable sidecar behavior or not. By default pod specs are merged into main application container.
-| `false`
+| *sidecar*
+| Whether to enable sidecar behavior or not. By default pod specs are merged into main application container. Defaults to `false`
+| `jkube.sidecar`
 
-| *jkube.skipHealthCheck*
-| Whether to skip health checks addition in generated resources or not.
-| `false`
+| *skipHealthCheck*
+| Whether to skip health checks addition in generated resources or not. Defaults to `false`
+| `jkube.skipHealthCheck`
 
-| *jkube.workDir*
-| The JKube working directory
-| `${project.build.directory}/jkube`
+| *workDir*
+| The JKube working directory. Defaults to `${project.build.directory}/jkube`
+| `jkube.workDir`
 
-| *jkube.environment*
-| Environment name where resources are placed. For example, if you set this property to dev and resourceDir is the default one, plugin will look at `src/main/jkube/dev`.
-| `NULL`
+| *environment*
+| Environment name where resources are placed. For example, if you set this property to dev and resourceDir is the default one, plugin will look at `src/main/jkube/dev`. Defaults to `null`
+| `jkube.environment`
 
-| *jkube.useProjectClassPath*
-| Should we use the project's compile time classpath to scan for additional enrichers/generators?
-| `false`
+| *useProjectClassPath*
+| Should we use the project's compile time classpath to scan for additional enrichers/generators. Defaults to `false`
+| `jkube.useProjectClassPath`
 
-| *jkube.resourceDir*
-| Folder where to find project specific files
-| `${basedir}/src/main/jkube`
+| *resourceDir*
+| Folder where to find project specific files. Defaults to `${basedir}/src/main/jkube`
+| `jkube.resourceDir`
 
-| *jkube.targetDir*
-| The generated Kubernetes manifests
-| `${project.build.outputDirectory}/META-INF/jkube`
+| *targetDir*
+| The generated Kubernetes manifests. Defaults to `${project.build.outputDirectory}/META-INF/jkube`
+| `jkube.targetDir`
 
+| *resourceType*
+| The artifact type for attaching the generated resource file to the project. Can be either 'json' or 'yaml'. Defaults to `yaml`
+| `jkube.resourceType`
+
+| *mergeWithDekorate*
+| When resource generation is delegated to Dekorate, should JKube resources be merged with Dekorate's. Defaults to `false`
+| `jkube.mergeWithDekorate`
+
+| *skipResource*
+| Skip resource generation. Defaults to `false`
+| `jkube.skip.resource`
 |===
 
 [[Other-flags]]

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-debug.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-debug.adoc
@@ -80,3 +80,22 @@ The remote application will start only after a remote debugger is attached. You 
 
 WARNING: The `jkube.debug.suspend` flag will disable readiness probes in the Kubernetes deployment in order to start port-forwarding during the early phases of application startup
 
+[[Supported-Properties-Debug]]
+=== Supported Properties For Debug Goal
+
+.Options available with debug goal
+[cols="1.6.3"]
+|===
+| Element | Description | Property(System property or maven property)
+
+| *debugPort*
+| Default port available for debugging your application inside Kubernetes. Defaults to `5005`
+| `jkube.debug.port`
+
+| *debugSuspend*
+| Disables readiness probes in Kubernetes Deployment in order to start port forwarding during early phases of application startup. Defaults to `false`
+| `jkube.debug.suspend`
+|===
+
+[[Supported-Properties]]
+

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-log.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-log.adoc
@@ -38,3 +38,24 @@ If your pod has multiple containers you can configure the container name to log 
 ----
 mvn k8s:log -Djkube.log.container=foo
 ----
+
+[[Supported-Properties-Log]]
+=== Supported Properties for Log goal
+
+.Options available with log goal
+[cols="1.6.3"]
+|===
+| Element | Description | Property(System property or Maven property)
+
+| *logFollow*
+| Get follow logs for your application inside Kubernetes. Defaults to `true`
+| `jkube.log.follow`
+
+| *logContainer*
+| Get logs of some specific container inside your application Deployment. Defaults to `null`
+| `jkube.log.container`
+
+| *logPod*
+| Get logs of some specific pod inside your application Deployment. Defaults to `null`
+| `jkube.log.pod`
+|===

--- a/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-watch.adoc
+++ b/kubernetes-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-watch.adoc
@@ -106,4 +106,17 @@ mvn package
 The watcher will detect that the binary artifact has changed and will first rebuild the docker image,
 then start a redeploy of the Kubernetes pod.
 
-It uses the watch feature of the https://dmp.jkube.io/#docker:watch[docker-maven-plugin] under the hood.
+It uses the watch feature of the https://dmp.fabric8.io/#docker:watch[docker-maven-plugin] under the hood.
+[[Supported-Properties-Watch]]
+=== Supported Properties for Watch goal
+
+.Options available with watch goal
+[cols="1.6.3"]
+|===
+| Element | Description | Property(System property or Maven property)
+
+| *kubernetesManifest*
+| The generated kubernetes YAML file. Defaults to `${basedir}/target/classes/META-INF/jkube/kubernetes.yml`
+| `jkube.kubernetesManifest`
+
+|===

--- a/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java
+++ b/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java
@@ -109,12 +109,6 @@ public class ResourceMojo extends AbstractJKubeMojo {
     private File resourceDir;
 
     /**
-     * Folder where to find project specific files
-     */
-    @Parameter(property = "jkube.resourceDirOpenShiftOverride", defaultValue = "${basedir}/src/main/jkube-openshift-override")
-    private File resourceDirOpenShiftOverride;
-
-    /**
      * Environment name where resources are placed. For example, if you set this property to dev and resourceDir is the default one, plugin will look at src/main/jkube/dev
      * Same applies for resourceDirOpenShiftOverride property.
      */
@@ -175,10 +169,6 @@ public class ResourceMojo extends AbstractJKubeMojo {
      * Enricher specific configuration configuration given through
      * to the various enrichers.
      */
-
-    // Resource specific configuration for this plugin
-    @Parameter(property = "jkube.gitRemote")
-    private String gitRemote;
 
     @Parameter
     private ProcessorConfig enricher;
@@ -257,7 +247,7 @@ public class ResourceMojo extends AbstractJKubeMojo {
     @Parameter(property = "jkube.openshift.enrichAllWithImageChangeTrigger", defaultValue = "false")
     private Boolean erichAllWithImageChangeTrigger;
 
-    @Parameter(property = "docker.skip.resource", defaultValue = "false")
+    @Parameter(property = "jkube.skip.resource", defaultValue = "false")
     protected boolean skipResource;
 
     /**

--- a/kubernetes-maven-plugin/plugin/src/main/resources/META-INF/jkube/profiles-default.yml
+++ b/kubernetes-maven-plugin/plugin/src/main/resources/META-INF/jkube/profiles-default.yml
@@ -12,7 +12,7 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-# Default profiles delivered with f-m-p
+# Default profiles delivered with kmp
 
 # tag::default[]
 # Default profile which is always activated

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/_access.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/_access.adoc
@@ -16,4 +16,5 @@ be specified by the certPath or machine configuration, or by the
 
 == OpenShift and Kubernetes Access
 
-If no `DOCKER_HOST` is set and no unix socket could be accessed under `/var/run/docker.sock` then f-m-p checks whethe https://github.com/jkubeio/gojkube.gojkube. is in the path and uses `gojkube.docker-env` to get the connection parameter to the Docker host exposed by
+Plugin reads your `kubeconfig` file to read your Kubernetes/OpenShift configuration. By default, `kubeconfig` file is assumed to be either
+in `~/.kube/config` or using the environment variable `KUBECONFIG`.

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/_enricher.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/_enricher.adoc
@@ -173,7 +173,7 @@ This enricher is disabled by default. In order to use it, you must configure the
 <plugin>
   <groupId>org.eclipse.jkube</groupId>
   <artifactId>openshift-maven-plugin</artifactId>
-  <version>3.3.0</version>
+  <version>{version}</version>
   <executions>
     <execution>
       <goals>
@@ -308,10 +308,10 @@ These values can be configured by the enricher in the `kubernetes-maven-plugin` 
 <plugin>
   <groupId>org.eclipse.jkube</groupId>
   <artifactId>openshift-maven-plugin</artifactId>
-  <version>3.3.0</version>
+  <version>{version}</version>
   <executions>
     <execution>
-      <id>fmp</id>
+      <id>kmp</id>
         <goals>
           <goal>resource</goal>
           <goal>helm</goal>
@@ -364,10 +364,10 @@ These values can be configured by the enricher in the `kubernetes-maven-plugin` 
       <plugin>
         <groupId>org.eclipse.jkube</groupId>
         <artifactId>openshift-maven-plugin</artifactId>
-        <version>3.3.0</version>
+        <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -416,10 +416,10 @@ These values can be configured by the enricher in the `kubernetes-maven-plugin` 
       <plugin>
         <groupId>org.eclipse.jkube</groupId>
         <artifactId>openshift-maven-plugin</artifactId>
-        <version>3.3.0</version>
+        <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -468,10 +468,10 @@ These values can be configured by the enricher in the `kubernetes-maven-plugin` 
       <plugin>
         <groupId>org.eclipse.jkube</groupId>
         <artifactId>openshift-maven-plugin</artifactId>
-        <version>3.3.0</version>
+        <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -522,7 +522,7 @@ The minimal configuration to add health checks is the following:
         <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -569,7 +569,7 @@ You can provide two different configuration for the readiness and liveness check
         <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>
@@ -614,7 +614,7 @@ Shared (generic) configuration can be set outside of the specific configuration.
         <version>{version}</version>
         <executions>
           <execution>
-            <id>fmp</id>
+            <id>kmp</id>
             <goals>
               <goal>resource</goal>
               <goal>helm</goal>

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/_introduction.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/_introduction.adoc
@@ -2,7 +2,7 @@
 [[introduction]]
 = Introduction
 
-The *openshift-maven-plugin* (oc-m-p) brings your Java applications on to https://www.openshift.com/[Openshift].
+The *openshift-maven-plugin* (omp) brings your Java applications on to https://www.openshift.com/[Openshift].
 It provides a tight integration into http://maven.apache.org[Maven] and benefits from the build configuration already provided.
 This plugin focus on two tasks: _Building Docker images_ and _creating Kubernetes resource descriptors_.
 It can be configured very flexibly and supports multiple configuration models for creating: A _Zero-Config_ setup allows for a quick ramp-up with some opinionated defaults.
@@ -56,11 +56,11 @@ The following table gives an overview of the different models
 | Default <<enrichers, Enrichers>> will create a default _Service_ and _Deployment_ (_DeploymentConfig_ for OpenShift) when no other resource objects are provided. Depending on the image they can detect which port to expose in the service. As with Generators, Enrichers support a limited set of configuration options.
 
 | *XML configuration*
-| oc-m-p inherits the XML based configuration for building images from the https://jkubeio.github.io/docker-maven-plugin/#docker:build[docker-maven-plugin] and provides the same functionality. It supports an http://maven.apache.org/components/plugins/maven-assembly-plugin/assembly.html[assembly descriptor] for specifying the content of the Docker image.
+| omp inherits the XML based configuration for building images from the https://jkubeio.github.io/docker-maven-plugin/#docker:build[docker-maven-plugin] and provides the same functionality. It supports an http://maven.apache.org/components/plugins/maven-assembly-plugin/assembly.html[assembly descriptor] for specifying the content of the Docker image.
 | A subset of possible resource objects can be configured with a dedicated XML syntax. With a decent IDE you get autocompletion on most object and inline documentation for the available configuration elements. The provided configuration can be still enhanced by Enhancers which is useful for adding e.g. labels and annotations containing build or other information.
 
 | *Resource Fragments* and *Dockerfiles*
-| Similarly to the docker-maven-plugin, oc-m-p supports <<external-dockerfile,external Dockerfiles>> too, which are referenced from the plugin configuration.
+| Similarly to the docker-maven-plugin, omp supports <<external-dockerfile,external Dockerfiles>> too, which are referenced from the plugin configuration.
 | Resource descriptors can be provided as external YAML files which specify a skeleton. This skeleton is then filled by Enrichers which add labels and more. Maven properties within these files are resolved to thier values. With this model you can use every Kubernetes / OpenShift resource object with all their flexibility, but still get the benefit of adding build information.
 |===
 
@@ -144,7 +144,7 @@ mvn oc:resource oc:deploy
 
 By default a _Service_ and a _Deployment_ object pointing to the created Docker image is created. When running in OpenShift mode, a _Service_ and _DeploymentConfig_ which refers the _ImageStream_ created with `oc:build` will be installed.
 
-Of course you can bind all those jkube.goals to execution phases as well, so that they are called along with standard lifecycle goals like `install`. For example, to bind the building of the Kubernetes resource files and the Docker images, add the following goals to the execution of the f-m-p:
+Of course you can bind all those jkube.goals to execution phases as well, so that they are called along with standard lifecycle goals like `install`. For example, to bind the building of the Kubernetes resource files and the Docker images, add the following goals to the execution of the omp:
 
 .Example for lifecycle bindings
 [source, xml, indent=0]
@@ -204,7 +204,7 @@ The plugin configuration can be roughly divided into the following sections:
 * `<generator>` configures <<generators,generators>> which are responsible for creating images. Generators are used as an alternative to a dedicated `<images>` section.
 * `<enricher>` configures various aspects of <<enrichers,enrichers>> for creating or enhancing resource descriptors.
 
-A working example can be found in the  https://github.com/jkubeio/kubernetes-maven-plugin/tree/master/samples/xml-config[samples/xml-config] directory. An extract of the plugin configuration is shown below:
+A working example can be found in the  https://github.com/eclipse/jkube/tree/master/quickstarts/maven/xml-config[samples/xml-config] directory. An extract of the plugin configuration is shown below:
 
 .Example for an XML configuration
 [source,xml,indent=0,subs="verbatim,quotes,attributes"]
@@ -217,10 +217,11 @@ A working example can be found in the  https://github.com/jkubeio/kubernetes-mav
       <!-- "alias" is used to correlate to the containers in the pod spec -->
       <alias>camel-app</alias>
       <build>
-        <from>jkube.java</from>
+        <from>fabric8/java-centos-openjdk8-jre</from>
         <assembly>
-          <basedir>/deployments</basedir>
-          <descriptorRef>artifact-with-dependencies</descriptorRef>
+           <inline>
+              <baseDirectory>/deployments</baseDirectory>
+           </inline>
         </assembly>
         <env>
           <JAVA_LIB_DIR>/deployments</JAVA_LIB_DIR>
@@ -237,31 +238,10 @@ A working example can be found in the  https://github.com/jkubeio/kubernetes-mav
       </all>
     </labels>
 
-    <deployment> <!--4-->
-      <name>${project.artifactId}</name>
-      <replicas>1</replicas>
+    <replicas>2</replicas> <!--4-->
+    <controllerName>${project.artifactId}</controllerName> <!--5-->
 
-      <containers> <!--5-->
-        <container>
-          <alias>camel-app</alias> <!--6-->
-          <ports>
-            <port>8778</port>
-          </ports>
-          <mounts>
-            <scratch>/var/scratch</scratch>
-          </mounts>
-        </container>
-      </containers>
-
-      <volumes> <!--7-->
-        <volume>
-          <name>scratch</name>
-          <type>emptyDir</type>
-        </volume>
-      </volumes>
-    </deployment>
-
-    <services> <!--8-->
+    <services> <!--6-->
       <service>
         <name>camel-service</name>
         <headless>true</headless>
@@ -279,11 +259,9 @@ A working example can be found in the  https://github.com/jkubeio/kubernetes-mav
 <1> Standard docker-maven-plugin configuration for building one single Docker image
 <2> Kubernetes / OpenShift resources to create
 <3> Labels which should be applied globally to all resource objects
-<4> Definition of a http://kubernetes.io/docs/user-guide/deployments/[Deployment] to create
-<5> Containers to include in the deployment
-<6> An _alias_ is used to correlate a container's image with the image definition in the `<images>` section where each image carries an alias. Can be omitted if only a single image is used
-<7> http://kubernetes.io/docs/user-guide/volumes/[Volume] definitions used in a Deployment's _ReplicaSet_
-<8> One or more http://kubernetes.io/docs/user-guide/services/[Service] definitions.
+<4> Number of replicas desired
+<5> Name of controller created by plugin
+<6> One or more http://kubernetes.io/docs/user-guide/services/[Service] definitions.
 
 The XML resource configuration is based on plain Kubernetes resource objects. When targeting OpenShift, Kubernetes resource descriptors will be automatically converted to their OpenShift counterparts, e.g. a Kubernetes http://kubernetes.io/docs/user-guide/deployments/[Deployment] will be converted to an OpenShift https://docs.openshift.org/latest/architecture/core_concepts/deployments.html#deployments-and-deployment-configurations[DeploymentConfig].
 
@@ -293,7 +271,7 @@ The third configuration option is to use an external configuration in form of YA
 
 Note: In order to support simultaneously both OpenShift and Kubernetes, there is currently no way to specify OpenShift-only features this way, though this might change in future releases.
 
-Let's have a look at an example from https://github.com/rhuss/kubernetes-maven-plugin/tree/master/samples/external-resources[samples/external-resources]. This is a plain Spring Boot application, whose images are auto generated like in the <<zero-config, Zero-Config>> case. The resource fragments are in `src/main/jkube`.
+Let's have a look at an example from https://github.com/eclipse/jkube/tree/master/quickstarts/maven/external-resources[quickstarts/maven/external-resources]. This is a plain Spring Boot application, whose images are auto generated like in the <<zero-config, Zero-Config>> case. The resource fragments are in `src/main/jkube`.
 
 .Example fragment "deployment.yml"
 [source,yaml,indent=0,subs="verbatim,quotes,attributes"]

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/generator/_java_exec.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/generator/_java_exec.adoc
@@ -22,9 +22,9 @@ It will use the following base image by default, but as explained <<generator-op
 | | Docker Build | S2I Build | ImageStream
 
 | *Community*
-| `jkube.java-jboss-openjdk8-jdk`
-| `jkube.s2i-java`
-| `jkube.java`
+| `fabric8/java-jboss-openjdk8-jdk`
+| `fabric8/s2i-java`
+| `fabric8-java`
 
 | *Red Hat*
 | `jboss-fuse-6/fis-java-openshift`

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-apply.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-apply.adoc
@@ -8,3 +8,77 @@ This goal applies the resources created with <<oc:resource>> to a connected Kube
 ----
 mvn oc:apply
 ----
+
+[[Supported-Properties-Apply]]
+=== Supported Properties For Apply goal
+
+.Other options available with apply goal
+[cols="1.6.3"]
+|===
+| Element | Description | Property(System property or maven property)
+
+| *failOnError*
+| Should we fail the build if an apply fails, by default `true`
+| `jkube.deploy.failOnError`
+
+| *recreate*
+| Should we update resources by deleting them first and then creating them again, by default `false`
+| `jkube.recreate`
+
+| *openshiftManifest*
+| The generated kubernetes YAML file. Defaults to `${basedir}/target/classes/META-INF/jkube/openshift.yml`
+| `jkube.openshiftManifest`
+
+| *create*
+| Should we create new resources, defaults to `true`
+| `jkube.deploy.create`
+
+| *rolling*
+| Should we use rolling updates to apply changes, defaults to `false`
+| `jkube.rolling`
+
+| *servicesOnly*
+| In services only mode we only process services so that those can be recursively created/updated first before creating/updating any pods and replication controllers, defaults to `false`
+| `jkube.deploy.servicesOnly`
+
+| *ignoreServices*
+| Do we want to ignore services. This is particularly useful when in recreate mode to let you easily recreate all the ReplicationControllers and Pods but leave any service definitions alone to avoid changing the portalIP addresses and breaking existing pods using the service. Defaults to `false`
+| `jkube.deploy.ignoreServices`
+
+| *processTemplatesLocally*
+| Process templates locally in Java so that we can apply OpenShift templates on any Kubernetes environment, defaults to `false`
+| `jkube.deploy.processTemplatesLocally`
+
+| *deletePods*
+| Should we delete all the pods if we update a Replication Controller, defaults to `true`
+| `jkube.deploy.deletePods`
+
+| *ignoreRunningOAuthClients*
+| Do we want to ignore OAuthClients which are already running?. OAuthClients are shared across namespaces so we should not try to update or create/delete global oauth clients, defaults to `true`
+| `jkube.deploy.ignoreRunningOAuthClients`
+
+| *createExternalUrls*
+| Should we create external Ingress for any LoadBalancer Services which don't already have them. Defaults to `false`
+| `jkube.deploy.createExternalUrls`
+
+| *jsonLogDir*
+| The folder we should store any temporary json files or results. Defaults to `${basedir}/target/jkube/applyJson`
+| `jkube.deploy.jsonLogDir`
+
+| *waitSeconds*
+| How many seconds to wait for a URL to be generated for a service , defaults to `5`
+| `jkube.serviceUrl.waitSeconds`
+
+| *resourceDir*
+| Folder where to find project specific files, defaults to `${basedir}/src/main/jkube`
+| `jkube.resourceDir`
+
+| *environment*
+|  Environment name where resources are placed. For example, if you set this property to dev and resourceDir is the default one, jkube will look at `src/main/jkube/dev`. Defaults to `null`
+| `jkube.environment`
+
+| *skipApply*
+| Skip applying the resources, defaults to `false`
+| `jkube.skip.apply`
+
+|===

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/build/_jkube-resource.adoc
@@ -238,66 +238,78 @@ spec:
 
 In case both the label and the property have been set with conflicting values, precedence will be given to the property value, so if you set the label to `true` but set the property to `false` then no Route descriptor will be generated because precedence will be given to the property value.
 
-[[Other-flags]]
-=== Other flags
+[[Supported-Properties-Resource]]
+=== Supported Properties for Resource goal
 
-.Other options available with resource goal
+.Options available with resource goal
 [cols="1.6.3"]
 |===
-| Configuration | Description | Default
+| Element | Description | Property(System property or maven property)
 
-| *jkube.openshift.enableAutomaticTrigger*
-| If the value is set to `false` then automatic deployments would be disabled.
-| `true`
+| *enableAutomaticTrigger*
+| If the value is set to `false` then automatic deployments would be disabled. Defaults to `true`
+| `jkube.openshift.enableAutomaticTrigger`
 
-| *jkube.skipHealthCheck*
-| If the value is set to `true` then no readiness/liveness checks would be added to any containers.
-| `false`
+| *skipHealthCheck*
+| Whether to skip health checks addition in generated resources or not. Defaults to `false`
+| `jkube.skipHealthCheck`
 
-| *jkube.openshift.deployTimeoutSeconds*
-| The OpenShift deploy timeout in seconds.
-| 3600
+| *deployTimeoutSeconds*
+| The OpenShift deploy timeout in seconds. Defaults to `3600`
+| `jkube.openshift.deployTimeoutSeconds`
 
-| *jkube.openshift.imageChangeTrigger*
-| Add ImageChange triggers to DeploymentConfigs when on openshift.
-| `true`
+| *imageChangeTrigger*
+| Add ImageChange triggers to DeploymentConfigs when on openshift. Defaults to `true`
+| `jkube.openshift.imageChangeTrigger`
 
-| *jkube.profile*
-| Profile to use. A profile contains the enrichers and generators to use as well as their configuration. Profiles are looked up in the classpath and can be provided as yaml files.
-| `default`
+| *trimImageInContainerSpec*
+| If set to true it would set the container image reference to "", this is done to handle weird behavior of Openshift 3.7 in which subsequent rollouts lead to ImagePullErr. Defaults to `false`
+| `jkube.openshift.trimImageInContainerSpec`
 
-| *jkube.sidecar*
-| Whether to enable sidecar behavior or not. By default pod specs are merged into main application container.
-| `false`
+| *enrichAllWithImageChangeTrigger*
+| Adds ImageChangeTriggers with respect to all containers specified inside DeploymentConfig. Defaults to `false`
+| `jkube.openshift.enrichAllWithImageChangeTrigger`
 
-| *jkube.skipHealthCheck*
-| Whether to skip health checks addition in generated resources or not.
-| `false`
+| *profile*
+| Profile to use. A profile contains the enrichers and generators to use as well as their configuration. Profiles are looked up in the classpath and can be provided as yaml files. Defaults to `default`
+| `jkube.profile`
 
-| *jkube.workDir*
-| The JKube working directory
-| `${project.build.directory}/jkube`
+| *sidecar*
+| Whether to enable sidecar behavior or not. By default pod specs are merged into main application container. Defaults to `false`
+| `jkube.sidecar`
 
-| *jkube.environment*
-| Environment name where resources are placed. For example, if you set this property to dev and resourceDir is the default one, plugin will look at `src/main/jkube/dev`.
-| `NULL`
+| *workDir*
+| The JKube working directory. Defaults to `${project.build.directory}/jkube`
+| `jkube.workDir`
 
-| *jkube.useProjectClassPath*
-| Should we use the project's compile time classpath to scan for additional enrichers/generators?
-| `false`
+| *environment*
+| Environment name where resources are placed. For example, if you set this property to dev and resourceDir is the default one, plugin will look at `src/main/jkube/dev`. Defaults to `null`
+| `jkube.environment`
 
-| *jkube.resourceDir*
-| Folder where to find project specific files
-| `${basedir}/src/main/jkube`
+| *useProjectClassPath*
+| Should we use the project's compile time classpath to scan for additional enrichers/generators. Defaults to `false`
+| `jkube.useProjectClassPath`
 
-| *jkube.targetDir*
-| The generated Kubernetes manifests
-| `${project.build.outputDirectory}/META-INF/jkube`
+| *resourceDir*
+| Folder where to find project specific files. Defaults to `${basedir}/src/main/jkube`
+| `jkube.resourceDir`
 
-| *jkube.switchToDeployment*
-| Use Deployment instead of DeploymentConfig when on Openshift.
-| `false`
+| *targetDir*
+| The generated Kubernetes manifests. Defaults to `${project.build.outputDirectory}/META-INF/jkube`
+| `jkube.targetDir`
+
+| *resourceType*
+| The artifact type for attaching the generated resource file to the project. Can be either 'json' or 'yaml'. Defaults to `yaml`
+| `jkube.resourceType`
+
+| *mergeWithDekorate*
+| When resource generation is delegated to Dekorate, should JKube resources be merged with Dekorate's. Defaults to `false`
+| `jkube.mergeWithDekorate`
+
+| *skipResource*
+| Skip resource generation. Defaults to `false`
+| `jkube.skip.resource`
 
 |===
 
-[[Other-flags]]
+[[Supported-Properties]]

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-debug.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-debug.adoc
@@ -80,3 +80,19 @@ The remote application will start only after a remote debugger is attached. You 
 
 WARNING: The `jkube.debug.suspend` flag will disable readiness probes in the Kubernetes deployment in order to start port-forwarding during the early phases of application startup
 
+[[Supported-Properties-Debug]]
+=== Supported Properties For Debug Goal
+
+.Options available with debug goal
+[cols="1.6.3"]
+|===
+| Element | Description | Property(System property or maven property)
+
+| *debugPort*
+| Default port available for debugging your application inside Kubernetes. Defaults to `5005`
+| `jkube.debug.port`
+
+| *debugSuspend*
+| Disables readiness probes in Kubernetes Deployment in order to start port forwarding during early phases of application startup. Defaults to `false`
+| `jkube.debug.suspend`
+|===

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-log.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-log.adoc
@@ -38,3 +38,24 @@ If your pod has multiple containers you can configure the container name to log 
 ----
 mvn oc:log -Djkube.log.container=foo
 ----
+
+[[Supported-Properties-Log]]
+=== Supported Properties for Log goal
+
+.Options available with log goal
+[cols="1.6.3"]
+|===
+| Element | Description | Property(System property or Maven property)
+
+| *logFollow*
+| Get follow logs for your application inside Kubernetes. Defaults to `true`
+| `jkube.log.follow`
+
+| *logContainer*
+| Get logs of some specific container inside your application Deployment. Defaults to `null`
+| `jkube.log.container`
+
+| *logPod*
+| Get logs of some specific pod inside your application Deployment. Defaults to `null`
+| `jkube.log.pod`
+|===

--- a/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-watch.adoc
+++ b/openshift-maven-plugin/doc/src/main/asciidoc/inc/goals/develop/_jkube-watch.adoc
@@ -107,3 +107,17 @@ The watcher will detect that the binary artifact has changed and will first rebu
 then start a redeploy of the Kubernetes pod.
 
 It uses the watch feature of the https://dmp.jkube.io/#docker:watch[docker-maven-plugin] under the hood.
+
+[[Supported-Properties-Watch]]
+=== Supported Properties for Watch goal
+
+.Options available with watch goal
+[cols="1.6.3"]
+|===
+| Element | Description | Property(System property or Maven property)
+
+| *kubernetesManifest*
+| The generated kubernetes YAML file. Defaults to `${basedir}/target/classes/META-INF/jkube/kubernetes.yml`
+| `jkube.kubernetesManifest`
+
+|===

--- a/openshift-maven-plugin/plugin/src/main/resources/META-INF/jkube/profiles-default.yml
+++ b/openshift-maven-plugin/plugin/src/main/resources/META-INF/jkube/profiles-default.yml
@@ -12,7 +12,7 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-# Default profiles delivered with f-m-p
+# Default profiles delivered with omp
 
 # tag::default[]
 # Default profile which is always activated

